tasks/Functions-consumer/performManualUpkeep.js

```diff
@@ -16,22 +16,42 @@ task("functions-perform-upkeep", "Manually call performUpkeep in an Automation c
     // Call performUpkeep
     const performData = taskArgs.data ?? []
 
+    const autoConsumerContractFactory = await ethers.getContractFactory("AutomatedFunctionsConsumer")
+    const autoConsumerContract = await autoConsumerContractFactory.attach(taskArgs.contract)
+
+    const checkUpkeep = await autoConsumerContract.checkUpkeep(performData)
```

Comment:
Possibly a nit:  Given that the goal is to make sure that we do not re run automation when the sub has no balance could we not just
1) performUpkeep anyway and handle a reviert if checkUpkeep returns false (this logic is now in performupkeep)
2) console log the error (with string or without) if the functions request fails

Do we need to call checkUpkeep separately here?

Reply:
I updated the console.log message slightly, although I do think how it is currently implemented is how I intended.

Reply:
sorry @KuphJr  i commented over too many lines! I was wondering whether line 22 could just go straight to performUpkeep (and rely on its internal checkupkeep call).